vaadin-icons-flow-parent/vaadin-icons-flow-integration-tests/src/test/java/com/vaadin/flow/component/icon/tests/FontIconIT.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.icon.tests;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.AbstractComponentIT;
+
+@TestPath("vaadin-icons/font-icons")
+public class FontIconIT extends AbstractComponentIT {
+
+    @Before
+    public void init() {
+        open();
+    }
+
+    @Test
+    public void fontIconUsingFont() {
+        var icon = findElement(By.className("fa-user"));
+        // Get the ::before element's width and height separated by a space
+        var beforeBounds = executeScript(
+                """
+                        const { width, height } = window.getComputedStyle(arguments[0], '::before');
+                        return `${width} ${height}`;
+                        """,
+                icon);
+
+        // Expect the icon size to match FontAwesome "user" icon's size
+        Assert.assertEquals("21px 24px", beforeBounds);
```

Comment:
Just a question / note: Ensures the Vaadin Icon that all icon's have the same width or has the Developer apply the fa-fw class name? (e.g. fa-exclamation is really narrow as example) 

How does some apply other font awesome related styles like fa-2xl or fa-rotate? Can those be directly applied to the Vaadin Icon?

Reply:
The `FontIcon` is sized the same as the regular `Icon` component, so by default, it is 24px/24px (`--lumo-icon-size-m`) and it's size can be changed with `setSize()`.

A font icon inside the `<vaadin-icon>` element will by default take 100% of the host's height, but its width is only determined by the icon's own aspect ratio (in the `fa-user`'s case, the width inside a 24px high icon becomes 21px etc). The icon is, by default, horizontally centered inside the host.

I added an outline around the `<vaadin-icon>` elements in the following examples:

![Screenshot 2023-08-30 at 9 45 27](https://github.com/vaadin/flow-components/assets/1222264/030a5000-5891-4ab2-b44b-edddaf27c593)

The plan is not to support all of FontAwesome's features specifically, but many of the utility classes still work fine, and they can be passed to the `FontIcon`'s constructor, or to `setFont()` method.

![Screenshot 2023-08-30 at 9 46 49](https://github.com/vaadin/flow-components/assets/1222264/eba7e1cb-9153-40f2-898c-c1fc0e84bb88)

![Screenshot 2023-08-30 at 9 47 23](https://github.com/vaadin/flow-components/assets/1222264/b09f4387-8c40-40dd-a5b5-c016a066a14c)

Some features, such as the sizing classes (`fa-xs`, `fa-4x`) don't work, see https://github.com/vaadin/web-components/pull/6418. You can instead size the `FontIcon` the same as the `Icon` component.

![Screenshot 2023-08-30 at 10 03 35](https://github.com/vaadin/flow-components/assets/1222264/74d460f1-f4fd-4a45-81c6-25f14a333462)

Reply:
Thanks for the comment! This are all really valuable information for the docs later on :)